string shape
string color
int32 cx
int32 cy
float32 area